### PR TITLE
Restructure CSVY Writer from Custom Abundance Widget 

### PR DESCRIPTION
**Pull request by rohithvarma3000**
### :pencil: Description

Created a new submodule called writers in io and put the CSVY Writer in Custom Abundance Widget in this new submodule.

**Type:** :rocket: `restructure` 

Write a complete description of your changes, including the necessary context or any piece of information required to understand your work.

Also, link issues affected by this pull request by using the keywords: `close`, `closes`, `closed`, `fix`, `fixes`, `fixed`, `resolve`, `resolves` or `resolved`. 

Addresses #1927 and a continuation of PR #1937 .

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [x] Other method (describe)
- [ ] My changes can't be tested (explain why)
Manual import and testing on examples.

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by andrewfullard** [comment]
To be clear: this just moves the functions to another file, correct?

**Comment by rohithvarma3000** [comment]
I have made all updates as per the discussions. Please have another look into the PR

File: .mailmap [hunk 1]
```diff
@@ -204,3 +204,7 @@ Wolfgang Kerzendorf <wkerzendorf@gmail.com> Wolfgang Kerzendorf <wkerzend@Wolfga
 
 Youssef Eweis <joey070@gmail.com>
 Youssef Eweis <joey070@gmail.com> Youssef Eweis <Youssef15015@users.noreply.github.com>
+
+Rohith Varma Buddaraju <rohith.varma.buddaraju@gmail.com>
+Rohith Varma Buddaraju <rohith.varma.buddaraju@gmail.com> rohithvarma3000 <rohith.varma.buddaraju@gmail.com>
+Rohith Varma Buddaraju <rohith.varma.buddaraju@gmail.com> Rohith <rohith.varma.buddaraju@gmail.com>
```
Comment (epassaro) [inline .mailmap:210]: you need to edit `.mailmap` on just one of your PRs!

**Review by andrewfullard — APPROVED** [review]

File: .mailmap (hunk 1 above)
Comment (rohithvarma3000) [reply .mailmap:210]: Ok. Once one PR is merged, I will make the necessary changes.

**Comment by andrewfullard** [comment]
Okay you can remove the mailmap edit from this PR. May need a rebase.

**Comment by Rodot-** [comment]
@andrewfullard @epassaro is this still active or should we close it?

**Pull request closed without merging**